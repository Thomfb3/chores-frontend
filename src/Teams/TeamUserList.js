import React, { useContext } from "react";
import TeamUserCard from "./TeamUserCard"
import LoadingSpinner from "../common/LoadingSpinner";
import UserContext from "../auth/UserContext";


function TeamUserList() {
  console.debug("TeamUserList");
  const { currentUser, currentTeam, currentTeamUsers } = useContext(UserContext);

  if (!currentTeamUsers) return <LoadingSpinner />;
  currentTeamUsers.sort((a, b) => (a.points < b.points && 1) || -1);

  return (
    <div className="TeamUserList">
      <div className="TeamUserList__title">
        <h3 className="TeamUserList__list-title">Team {currentTeam.name}</h3>
        <div className="TeamUserList__divider"></div>
      </div>
      {currentTeamUsers.length ? (
        <div >
          {currentTeamUsers.map((u,idx)=> (
            <TeamUserCard
                id={u._id}
                key={u._id}
                isCurrentUser={u.username === currentUser.username}
                profileImage={u.profileImage}
                username={u.username}
                firstName={u.firstName}
                isAdmin={u.role === "admin"}
                currentPoints={u.currentPoints}
                allTimePoints={u.allTimePoints}
                position={idx+1}
            />
          ))}
        </div>
      ) : (
          <p>No Chores found!</p>
        )}
    </div>
  );
};

export default TeamUserList;