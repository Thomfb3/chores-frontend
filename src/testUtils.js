import React from "react";
import UserContext from "./auth/UserContext";
import ChoreCommentContext from "./comments/ChoreCommentContext";

const demoUser = {
  _id: "userid1",
  username: "testuser1",
  firstName: "testfirst1",
  lastName: "testlast1",
  profileImage: "defaultProfile.jpg",
  allTimePoints: 1000,
  currentPoints: 200,
  role: "admin"
};

const testTeam = {
  _id: "testTeam",
  name: "testTeam"
};

const testTeamUsers = [
  {
    _id: "userid1",
    username: "testuser1",
    firstName: "testfirst1",
    lastName: "testlast1",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "admin"
  },
  {
    _id: "userid2",
    username: "testuser2",
    firstName: "testfirst2",
    lastName: "testlast2",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid3",
    username: "testuser3",
    firstName: "testfirst3",
    lastName: "testlast3",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid4",
    username: "testuser4",
    firstName: "testfirst4",
    lastName: "testlast4",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  }
];

let choreComments = [];

const UserProvider =
  ({ children,
    currentUser = demoUser,
    currentTeam = testTeam,
    currentTeamUsers = testTeamUsers,
    isAdmin = false
  }) => (
      <UserContext.Provider value={{ currentUser, currentTeam, currentTeamUsers }}>
        {children}
      </UserContext.Provider>
    );


const CommentProvider =
  ({ children,
    currentChoreComments = choreComments
  }) => (
      <ChoreCommentContext.Provider value={{ currentChoreComments }}>
        {children}
      </ChoreCommentContext.Provider>
    );

export { UserProvider, CommentProvider };

