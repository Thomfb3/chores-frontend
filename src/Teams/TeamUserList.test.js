import React from "react";
import { render } from "@testing-library/react";
import TeamUserList from "./TeamUserList";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
    render(
        <UserProvider>
            <TeamUserList />
        </UserProvider>,
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <UserProvider>
            <TeamUserList />
        </UserProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
});
