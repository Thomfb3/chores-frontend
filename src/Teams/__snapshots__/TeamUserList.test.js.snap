// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TeamUserList"
  >
    <div
      class="TeamUserList__title"
    >
      <h3
        class="TeamUserList__list-title"
      >
        Team testTeam
      </h3>
      <div
        class="TeamUserList__divider"
      />
    </div>
    <div>
      <div
        class="TeamUserCard"
      >
        <div
          class="TeamUserCard__container"
        >
          <div
            class="TeamUserCard__inner-container TeamUserCard__inner-container--align-center"
          >
            <div
              class="TeamUserCard__position-container"
            >
              <div
                class="TeamUserCard__position TeamUserCard__first"
              >
                1
              </div>
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-52v5ae-MuiSvgIcon-root"
              data-testid="StarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
              />
            </svg>
            <div
              class="MuiAvatar-root MuiAvatar-circular Profile__avatar--component css-6w89ng-MuiAvatar-root"
            >
              <img
                alt="default-profile-pic.gif"
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="default-profile-pic.gif"
              />
            </div>
            <div
              class="TeamUserCard__name-grow"
            >
              <div>
                <p
                  class="TeamUserCard__firstname"
                >
                  testfirst1
                </p>
                <p
                  class="TeamUserCard__username"
                >
                  testuser1
                </p>
              </div>
              <div
                class="TeamUserCard__titles"
              >
                <p>
                  (Me)
                </p>
                <p>
                  (Team Manager)
                </p>
              </div>
            </div>
            <div
              class="TeamUserCard__all-points"
            >
              <p
                class="TeamUserCard__all-points--label TeamUserCard__points-first"
              >
                All Time Points
              </p>
              <p
                class="TeamUserCard__all-points--points TeamUserCard__points-first"
              >
                1000 pts
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="TeamUserCard"
      >
        <div
          class="TeamUserCard__container"
        >
          <div
            class="TeamUserCard__inner-container TeamUserCard__inner-container--align-center"
          >
            <div
              class="TeamUserCard__position-container"
            >
              <div
                class="TeamUserCard__position TeamUserCard__second"
              >
                2
              </div>
            </div>
            <div
              class="MuiAvatar-root MuiAvatar-circular Profile__avatar--component css-6w89ng-MuiAvatar-root"
            >
              <img
                alt="default-profile-pic.gif"
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="default-profile-pic.gif"
              />
            </div>
            <div
              class="TeamUserCard__name-grow"
            >
              <div>
                <p
                  class="TeamUserCard__firstname"
                >
                  testfirst2
                </p>
                <p
                  class="TeamUserCard__username"
                >
                  testuser2
                </p>
              </div>
              <div
                class="TeamUserCard__titles"
              />
            </div>
            <div
              class="TeamUserCard__all-points"
            >
              <p
                class="TeamUserCard__all-points--label TeamUserCard__points-second"
              >
                All Time Points
              </p>
              <p
                class="TeamUserCard__all-points--points TeamUserCard__points-second"
              >
                1000 pts
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="TeamUserCard"
      >
        <div
          class="TeamUserCard__container"
        >
          <div
            class="TeamUserCard__inner-container TeamUserCard__inner-container--align-center"
          >
            <div
              class="TeamUserCard__position-container"
            >
              <div
                class="TeamUserCard__position TeamUserCard__third"
              >
                3
              </div>
            </div>
            <div
              class="MuiAvatar-root MuiAvatar-circular Profile__avatar--component css-6w89ng-MuiAvatar-root"
            >
              <img
                alt="default-profile-pic.gif"
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="default-profile-pic.gif"
              />
            </div>
            <div
              class="TeamUserCard__name-grow"
            >
              <div>
                <p
                  class="TeamUserCard__firstname"
                >
                  testfirst3
                </p>
                <p
                  class="TeamUserCard__username"
                >
                  testuser3
                </p>
              </div>
              <div
                class="TeamUserCard__titles"
              />
            </div>
            <div
              class="TeamUserCard__all-points"
            >
              <p
                class="TeamUserCard__all-points--label TeamUserCard__points-third"
              >
                All Time Points
              </p>
              <p
                class="TeamUserCard__all-points--points TeamUserCard__points-third"
              >
                1000 pts
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="TeamUserCard"
      >
        <div
          class="TeamUserCard__container"
        >
          <div
            class="TeamUserCard__inner-container TeamUserCard__inner-container--align-center"
          >
            <div
              class="TeamUserCard__position-container"
            >
              <div
                class="TeamUserCard__position TeamUserCard__other-number--th"
              >
                4
              </div>
            </div>
            <div
              class="MuiAvatar-root MuiAvatar-circular Profile__avatar--component css-6w89ng-MuiAvatar-root"
            >
              <img
                alt="default-profile-pic.gif"
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="default-profile-pic.gif"
              />
            </div>
            <div
              class="TeamUserCard__name-grow"
            >
              <div>
                <p
                  class="TeamUserCard__firstname"
                >
                  testfirst4
                </p>
                <p
                  class="TeamUserCard__username"
                >
                  testuser4
                </p>
              </div>
              <div
                class="TeamUserCard__titles"
              />
            </div>
            <div
              class="TeamUserCard__all-points"
            >
              <p
                class="TeamUserCard__all-points--label TeamUserCard__points-other-number--th"
              >
                All Time Points
              </p>
              <p
                class="TeamUserCard__all-points--points TeamUserCard__points-other-number--th"
              >
                1000 pts
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
